libs/natparse/src/main/java/org/amshove/natparse/parsing/OutputStatementNode.java

```diff
@@ -0,0 +1,58 @@
+package org.amshove.natparse.parsing;
+
+import org.amshove.natparse.ReadOnlyList;
+import org.amshove.natparse.lexing.SyntaxToken;
+import org.amshove.natparse.natural.*;
+import org.amshove.natparse.natural.output.IOutputElementNode;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+class OutputStatementNode extends StatementNode implements IWriteNode, IPrintNode, IDisplayNode, ICanSetReportSpecification
```

Comment:
Remove the `IWriteNode`, `IPrintNode` and `IDisplayNode` here and add the `implements` at their corresponding class.

Otherwise each `OutputStatementNode` (e.g. a `WRITE`) is also a `IDisplayNode`

Other than that the approach seems feasable, as long as they have the same possible attributes all the time :)